packages/dart_frog_cli/test/src/commands/dev/dev_test.dart

```diff
@@ -181,6 +189,43 @@ void main() {
       verify(() => logger.err('oops')).called(1);
     });
 
+    test('fails if hostname is invalid', () async {
+      when(() => runner.start()).thenAnswer((_) => Future.value());
+      when(() => runner.exitCode).thenAnswer(
+        (_) => Future.value(ExitCode.success),
+      );
+
+      when(() => argResults['hostname']).thenReturn('ticarica');
+      when(() => argResults['port']).thenReturn('1234');
+      when(() => argResults['dart-vm-service-port']).thenReturn('5678');
+
+      final cwd = Directory.systemTemp;
+
+      final command = DevCommand(
+        generator: (_) async => generator,
+        ensureRuntimeCompatibility: (_) {},
+        devServerRunnerBuilder: ({
+          required logger,
+          required port,
+          required address,
+          required devServerBundleGenerator,
+          required dartVmServicePort,
+          required workingDirectory,
+          void Function()? onHotReloadEnabled,
+        }) {
+          return runner;
+        },
+        logger: logger,
+      )
+        ..testStdin = stdin
+        ..testArgResults = argResults
+        ..testCwd = cwd;
+
+      await expectLater(command.run(), completion(ExitCode.software.code));
```

Comment:
Should we also verify the logged error message?

If you modify the logged error in line 127 dev.dart all tests will pass.